mipengine/controller/node_landscape_aggregator.py

```diff
@@ -274,167 +327,206 @@ def get_node_specific_datasets(
         )
 
 
-def _get_datasets_per_node(
-    nodes: Dict[str, NodeInfo],
-) -> Dict[str, Dict[str, Dict[str, str]]]:
-    datasets_per_node = {}
-    for node_id, node_info in nodes.items():
-        node_socket_addr = _get_node_socket_addr(node_info)
-        datasets_per_data_model = _get_node_datasets_per_data_model(node_socket_addr)
-        if datasets_per_data_model:
-            datasets_per_node[node_info.id] = datasets_per_data_model
-    return datasets_per_node
-
-
-def remove_duplicated_datasets(datasets_per_node):
-    aggregated_datasets = {}
-    for node_id, datasets_per_data_model in datasets_per_node.items():
-        for data_model, datasets in datasets_per_data_model.items():
-            if data_model not in aggregated_datasets:
-                aggregated_datasets[data_model] = []
-            aggregated_datasets[data_model].extend(datasets.keys())
-
-    duplicated_datasets = {
-        data_model: [
-            item for item, count in Counter(dataset_names).items() if count > 1
-        ]
-        for data_model, dataset_names in aggregated_datasets.items()
-    }
+def _federation_nodes_metadata_fetching() -> Tuple[
+    List[NodeInfo],
+    DataModelsMetadataPerNode,
+]:
+    """
+    Returns a list of all the nodes in the federation and their metadata (data_models, datasets, cdes).
+    """
+    nodes_addresses = get_nodes_addresses()
+    nodes_info = _get_nodes_info(nodes_addresses)
+    local_nodes = [
+        node_info for node_info in nodes_info if node_info.role == NodeRole.LOCALNODE
+    ]
+    data_models_metadata_per_node = _get_data_models_metadata_per_node(local_nodes)
+    return nodes_info, data_models_metadata_per_node
 
-    datasets_per_node_without_duplicates = {
-        node_id: {
-            data_model: {
-                dataset_name: dataset_label
-                for dataset_name, dataset_label in datasets.items()
-                if dataset_name not in duplicated_datasets[data_model]
-            }
-            for data_model, datasets in datasets_per_data_model.items()
-        }
-        for node_id, datasets_per_data_model in datasets_per_node.items()
-    }
 
-    _log_duplicated_datasets_per_node(
-        datasets_per_node, datasets_per_node_without_duplicates
+def _data_model_registry_data_cruncing(
```

Comment:
the name of the function does not really give information on what it is doing, every line of code ever written does some data crunching..